### DISCUSSION TITLE
Filtering array content using contains with a variable value

### DISCUSSION THREAD

**Discussion by donguinesjay** [statamic/cms#5416] [Q&A] [answered]
Hi is there a way to make a condition to check if an array contains a specific variable value?

eg. `{{ if { arr_data | in_array:'{title}' }  }}`

I want to check if the value of title variable is inside the arr_data.

**Comment by ryanmitchell** [comment]
Did you try:

{{ if arr_data | in_array:title }}

I believe it checks whether a variable with that name exists when you pass a string.

**Reply by donguinesjay** [reply]
I did try that. But it always says true. As you can see the `title` variable has a string value. But it always says true to all `title` value even though the its not actually in the `arr_data`

**Reply by donguinesjay** [reply]
Here's the content of arr_data:
![image](https://user-images.githubusercontent.com/29100485/157040683-3711de84-ed83-43ef-8786-7e21c7b9bee6.png)

And here's all the title value:
![image](https://user-images.githubusercontent.com/29100485/157040833-a0f6cfa5-96a9-4231-8dd8-d6abb8e8cc4c.png)

I want to only show the existing title that are inside the `arr_data`. But as you can see, the condition is always saying true, Thus all the `title` value is showing even though its not existing in the `arr_data`

Before the `What's on` it should say `Not Found!` as its not existing inside the arr_data.
![image](https://user-images.githubusercontent.com/29100485/157041435-ed240ecc-6e25-4915-bdfc-ec74b140a9f3.png)

**Reply by jasonvarga** [reply]
The `in_array` is backwards from what you want. The value is the needle and you pass the haystack array.
You want the `contains` modifier. The value is the array haystack and you pass in the needle.

**Comment by jasonvarga** [comment, marked as answer]
Try `{{ if arr | contains:title }}`

**Reply by donguinesjay** [reply]
Thanks, It's working now!
